src/app/product.service.ts

```diff
@@ -0,0 +1,41 @@
+import { map } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductService {
+  constructor(private db: AngularFireDatabase) {}
+
+  create(product) {
+    return this.db.list('/products').push(product);
+  }
+
+  getAll() {
```

Comment:
stick to a consistent naming convention for your services. I would rename this function contract to: 

getProductsObservable(): Observable<Product[]>